_scripts/credentials.json

```diff
@@ -0,0 +1 @@
+{"installed":{"client_id":"557960053095-9ir0ene9v7c825la7trc10e0scblgf6e.apps.googleusercontent.com","project_id":"quickstart-1585151530357","auth_uri":"https://accounts.google.com/o/oauth2/auth","token_uri":"https://oauth2.googleapis.com/token","auth_provider_x509_cert_url":"https://www.googleapis.com/oauth2/v1/certs","client_secret":"Yd5KI3YVjFBI0Zm66JgBWo9p","redirect_uris":["urn:ietf:wg:oauth:2.0:oob","http://localhost"]}}
```

Comment:
These secrets shouldn't be here, should they?

Reply:
Hmm, I'm not sure how can we avoid not having client_id and client_secret here. This tells google which application (and hence which scopes, in this case spreadsheet read access) it should get token for. It seems that in various examples around the web, these two attributes are part of the client-code. You will still need to exchange these credentials for temporary token when running the app. 

I will try to do some more research to see what are the risks of leaking client_secret or whether there are some best-practices around using oauth in code that lives in public repositories.

Reply:
Either way, this client_secret is already compromised ;)

Reply:
Can't we have a "Anyone with the link can read" version of the spreadsheet? So we can pull the data without the need for auth. It's data that we _want to publish_ anyway, so no harm in making the spreadsheet itself public.

If that's not an option, then I'd suggest reading this values from environment variables, and (once we decide were to run this from) configure those. I'd start trying to run this from Github Actions (for consistency's sake), but can try alternatives too.

Reply:
After looking into this some more I think that having readable by world spreadsheet would eliminate the need for this tedious auth process, one thing we may want to consider here is redacting out the submitter responses.

One thing that comes to mind, and makes the review process just slightly more clunky, would be to have reviewers copy the relevant fields to another world-readable spreadsheet that will be the ground truth for automation. Adds extra step & extra spreadsheet, but eliminates the code complexity. There doesn't seem to be a lot of incoming resources at the moment so this process seems okay to me for now.

What do you think?

Reply:
Eliminating the oauth will also make it possible to run this automation periodically and w/o human interaction. OAuth still needs human interaction to fetch the auth token from the client secrets.